standards/bulk.md

```diff
@@ -0,0 +1,134 @@
+            "entity": {
+                ... MATCHING ENTITY...                                      // must match entity schema resource from the collection
+            }                                                               // schema on entity is static and not dynamic
+        },
+    ]
+}
+```
+
+- Bulk operations **MUST** include a response body schema following:
+```json
+{
+    "status": "SUCCEEDED" | "FAILED" | "PARTIAL",                           // overall status of bulk operation, partial indicating 
+    "operations": [                                                         // there are some operations that failed and succeeded
+        {
+            "ordinal": 0,                                                   // position in original ordered operation collection
+            "action": "CREATE" | "UPDATE" | "CREATE_UPDATE" | "DELETE",     // repeat action type
+            "id": "string" | null,                                          // the associated id of the entity, if available
+            "ref": "sps-ref" | null,                                        // the associated sps-ref URN entity, if applicable
+            "status": "SUCCEEDED" | "FAILED",                               // status of individual operation
+            "detail": "string",                                             // indicates details on the operation result, such as a error message.
+        }
+    ]
+}
+```
+
+The following example demonstrates various operations in a bulk request and response:
+
+```
+// REQUEST
+PATCH /articles
+Content-Type: application/json
+{
+    "operations": [
+        {
+            "action": "CREATE_UPDATE"
+            "ifMatch": "33a64df551425fcc55e4d42a148795d9f25f89d4",
+            "entity": {
+                "id": "bfd8f0c0-be67-4f81-bf82-e55e552609f4",
+                "name": "my name",
+                "description": "my description"
+            }
+        },
+        {
+            "action": "CREATE"  
+            "entity": {
+                "id": null,
+                "name": "my name",
+                "description": "my description"
+            }
+        },
+        {
+            "action": "DELETE" 
+            "ifMatch": "44a64df551425fcc55e4d42a148795d9f25f89c5",     
+            "entity": {
+                "id": "d9bd5d91-fc25-4410-ae42-c8f631e8e9ff",
+                "name": null,
+                "description": null
+            }
+        },
+    ]
+}
+ 
+// RESPONSE
+200 OK
+Content-Type: application/json
+{
+    "status": "PARTIAL",                                 
+    "operations": [
+        {
+            "ordinal": 0,
+            "action": "CREATE_UPDATE",
+            "id": "bfd8f0c0-be67-4f81-bf82-e55e552609f4",
+            "ref": "sps:...",
+            "status": "SUCCEEDED",                         
+            "detail": "Article was updated.",  
```

Comment:
Should this detail actually match something like the error format to allow for a list of possible issues on creation? should there be another "error" field that is a representation of the error format in its verbose details?

Reply:
I like the idea of having a full error object in here to have symmetry with calling things individually. But I'm not really sure how elegant that would be in terms of response schema here, though. By allowing partial success the schema has to kind of be lowest common denominator and have a place for both success and failure that co-exist but are probably mutually-exclusive on a particular item in the list. 🤔

Reply:
Should success and error detail be separate?

Reply:
going to combine into one detail with a complex object.